src/documents.c

```diff
@@ -39,17 +39,23 @@
 static XdpDbusDocuments *documents = NULL;
 static char *documents_mountpoint = NULL;
 
-void
-init_document_proxy (GDBusConnection *connection)
+gboolean
+init_document_proxy (GDBusConnection  *connection,
+                     GError          **error)
 {
   documents = xdp_dbus_documents_proxy_new_sync (connection, 0,
                                                  "org.freedesktop.portal.Documents",
                                                  "/org/freedesktop/portal/documents",
-                                                 NULL, NULL);
+                                                 NULL, error);
+  if (!documents)
+    return FALSE;
+
   xdp_dbus_documents_call_get_mount_point_sync (documents,
                                                 &documents_mountpoint,
                                                 NULL, NULL);
```

Comment:
Not a merge blocker for this change IMO, but should this check for errors too?

Reply:
There is only one place that doesn't check for it to be NULL and that's in `register_document`. Not sure if that should also handle NULL or if we should exit early like in the other cases.

Reply:
I don't know the documents portal well enough to say. Definitely a job for a separate PR, either way.